extensions/src/poseHand/index.ts

```diff
@@ -0,0 +1,316 @@
+                this.runtime.emit(this.runtime.constructor.PERIPHERAL_DISCONNECTED);
+            }
+            */
+        }
+        const estimateThrottleTimeout = (+new Date() - time) / 4;
+        await new Promise(r => setTimeout(r, estimateThrottleTimeout));
+    }
+  }
+  
+  /**
+   * Estimates where the hand is on the video frame.
+   * @param imageElement
+   * @returns {Promise<AnnotatedPrediction[]>}
+   */
+  async estimateHandPoseOnImage(imageElement) {
+    const handModel = await this.getLoadedHandModel();
+    return await handModel.estimateHands(imageElement, {
+        flipHorizontal: false
+    });
+  }
+
+  /**
+   * Gets the hand model from handpose
+   * @returns
+   */
+  async getLoadedHandModel() {
+    if (!this._handModel) {
+        this._handModel = await handpose.load();
+    }
+    return this._handModel;
+  }
+
+  /**
+   * Turns the video camera off/on/on and flipped. This is called in the operation of videoToggleBlock
+   * @param state 
+   */
+  videoToggle (state: number) {
+    if (state === VideoState.OFF) {
+      this.runtime.ioDevices.video.disableVideo();
+    } 
+    else {
+      this.runtime.ioDevices.video.enableVideo();
+      // Mirror if state is ON. Do not mirror if state is ON_FLIPPED.
+      this.runtime.ioDevices.video.mirror = (state === VideoState.ON);
+    }
+  }
+
+  /**
+   * Sets the video's transparency. This is called in the operation of setVideoTransparencyBlock
+   * @param transparency 
+   */
+  setVideoTransparency (transparency: number) {
+    const trans = Math.max(Math.min(transparency,100), 0);
+    this.runtime.ioDevices.video.setPreviewGhost(trans);
+  }
+
+  /**
+   * Sets up the default settings for the extension. Gives information related to each of the extension's blocks
+   * @returns The extension's blocks
+   */
+  defineBlocks(): PoseHand["BlockDefinitions"] {
+    
+    /**
+     * Sets up the extension's default video settings
+     */
+    if (this.firstInstall) {
+      this.globalVideoState = VideoState.ON;
+      this.globalVideoTransparency = 50;
+      this.projectStarted();
+      this.firstInstall = false;
+      this._handModel = null;
+    }
+    
+    /**
+     * The options for each finger
+     * @type {Array}
+     */
+    const fingerOptions = 
+    [{text: "thumb", value: "thumb"}, {text: "index finger", value: "indexFinger"},
+    {text: "middle finger", value: "middleFinger"}, {text: "ring finger", value: "ringFinger"}, {text: "pinky finger", value: "pinky"}];
+
+    /**
+     * The options for the part of a finger
+     * @type {Array}
+     */
+    const partOfFingerOptions = [{text: "tip", value: 3}, {text: "first knuckle", value: 2},
+    {text: "second knuckle", value: 1}, {text: "base", value: 0}];
+
+    
+    type DefineGoToHandPart = DefineBlock<PoseHand, Blocks["goToHandPartBlock"]>;
+    const goToHandPartBlock: DefineGoToHandPart = () => ({
+      type: BlockType.Command,
+      args: [{type: ArgumentType.String, 
```

Comment:
This formatting looks a little wonky! Could you install this extension: 
https://marketplace.visualstudio.com/items?itemName=rvest.vs-code-prettier-eslint
which should help automatically format this file with consistent indents. 

It might automatically format when you save the file? If it doesn't let me know and we can look around how to get it working. The default settings should be good enough for us!